src/components/trip-planner/TripPlan.svelte

```diff
@@ -0,0 +1,205 @@
+<script>
+	import { debounce } from '$lib/utils';
+	import { createEventDispatcher, onMount } from 'svelte';
+	import TripPlanSearchField from './TripPlanSearchField.svelte';
+	import { error } from '@sveltejs/kit';
+	import { browser } from '$app/environment';
+	export let mapProvider;
+
+	let fromPlace = '';
+	let toPlace = '';
+	let fromResults = [];
+	let toResults = [];
+	let selectedFrom = null;
+	let selectedTo = null;
+	let isLoadingFrom = false;
+	let isLoadingTo = false;
+	let fromMarker;
+	let toMarker;
+	let loading = false;
+
+	const dispatch = createEventDispatcher();
+
+	const fetchLocationResults = debounce(async (query, isFrom) => {
+		if (isFrom) {
```

Comment:
I don't think we can simplify the code in this way, because with the current implementation, we need to keep isLoadingTo independent. If we simply assign isLoadingTo = !isFrom, it will set isLoadingTo to true whenever isFrom is false, which is not the intended behavior. Instead, we want each flag, isLoadingFrom and isLoadingTo, to reflect the loading state only when a specific query (either 'from' or 'to') is being fetched. This requires us to manage each flag separately based on the type of request.

Reply:
Yes of course